optimum/habana/transformers/generation/utils.py

```diff
@@ -140,7 +140,10 @@ def get_final_stopping_criteria(x):
     if isinstance(x, bool):
         return x
     elif torch.is_tensor(x):
-        return all(x)
+        if x.dim() > 0:
+            return all(x)
+        else:
+            return x
```

Comment:
```suggestion
        return x.all() if x.dim() > 0 else x
```

Use `.all()` If you want to check if all elements in the tensor are `True`. `all(x)` will fail for 2D tensors and the output is not a tensor, it is not matching with x